components/src/preact/genomeViewer/CDSPlot.tsx

```diff
@@ -0,0 +1,248 @@
+                        style={{
+                            left: `${leftPercent}%`,
+                            width: `${widthPercent}%`,
+                            top: '1.5rem',
+                            whiteSpace: 'nowrap',
+                        }}
+                    >
+                        {width >= averageWidth ? tick.start : ''}
+                    </div>
+                );
+            })}
+        </div>
+    );
+};
+
+function getTooltipPosition(cds_start: number, cds_end: number, genomeLength: number) {
+    const tooltipY = cds_start + (cds_end - cds_start) / 2 < genomeLength / 2 ? 'start' : 'end';
+    return `bottom-${tooltipY}` as const;
+}
+
+function getToolTipContent(cds: CDSFeature) {
+    const cdsLength = cds.positions.reduce((acc, pos) => acc + pos.end - pos.start, 0);
+    const cdsCoordinates = cds.positions.map((pos) => `${pos.start}-${pos.end}`).join(', ');
+    return (
+        <div>
+            <p>
+                <span className='font-bold'>{cds.label}</span>
+            </p>
+            <tbody>
+                <tr>
+                    <th className='font-medium'>CDS Length</th>
+                    <td>{cdsLength}</td>
+                </tr>
+                <tr>
+                    <th className='font-medium'>CDS Coordinates</th>
+                    <td>{cdsCoordinates}</td>
+                </tr>
+            </tbody>
+        </div>
+    );
+}
+
+interface CDSBarsProps {
+    gffData: CDSFeature[][];
+    zoomStart: number;
+    zoomEnd: number;
+}
+
+const CDSBars: FunctionComponent<CDSBarsProps> = (componentProps) => {
+    const { gffData, zoomStart, zoomEnd } = componentProps;
+    const visibleRegionLength = useMemo(() => zoomEnd - zoomStart, [zoomStart, zoomEnd]);
+    return (
+        <>
+            {gffData.map((data, list_id) => (
+                <div className={'w-full h-6 relative'} key={list_id}>
+                    {data.map((cds, idx) => (
+                        <Fragment key={idx}>
+                            {cds.positions.map((position, posIdx) => {
+                                const start = Math.max(position.start, zoomStart);
+                                const end = Math.min(position.end, zoomEnd);
+
+                                if (start >= end) {
+                                    return null;
+                                }
+
+                                const widthPercent = ((end - start) / visibleRegionLength) * 100;
+                                const leftPercent = ((start - zoomStart) / visibleRegionLength) * 100;
+                                const tooltipPosition = getTooltipPosition(start, end, visibleRegionLength);
+                                const tooltipContent = getToolTipContent(cds);
+                                return (
+                                    <Tooltip
+                                        content={tooltipContent}
```

Comment:
This is actually a component -> let's make it one: rename `getToolTipContent` -> `TooltipContent` and make the arguments of that function `props`, and:
```suggestion
                                
                                return (
                                    <Tooltip
                                        content={<TooltipContent ... />}
```